### PR TITLE
Add issue numbers to some unimplemented feature errors

### PR DESCRIPTION
**Pull request by apantel**
Partial fix for #42547
Release note (sql change): CockroachDB now provides a link to the
relevant github issue when clients attempt to use certain features that
are not yet implemented.

### CODE REVIEW

**Comment by knz** [comment]
Thanks for this.

1) please ensure the label `X-anchored-telemetry` is present on all the linked issues. This ensures that we don't mistakenly close the issue until we've checked that it's not referenced any more.

2) this is a user-facing change (UX improvement) so you can create a summary of the improvement in a release note.

**Review by apantel — COMMENTED** [review]
I added the label to the issues that didn't have it. Thanks!

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/42847#-:-Lv7T5YxBzec8Rz1CKEI:bi4rnjr)* status: :shipit: complete! 0 of 0 LGTMs obtained

**Comment by knz** [comment]
:+1:

for the release note try this:

```
Release note (sql change): CockroachDB now provides a link to
the relevant github issue when clients attempt to use certain features
that are not yet implemented.
```

(We don't yet have `ux improvement` as a category)

**Review by knz — APPROVED** [review]

File: pkg/sql/virtual_schema.go [hunk 1]
```diff
@@ -395,7 +395,7 @@ func (vs *VirtualSchemaHolder) getVirtualTableEntry(tn *tree.TableName) (virtual
 			return t, nil
 		}
 		if _, ok := db.allTableNames[tableName]; ok {
-			return virtualDefEntry{}, unimplemented.Newf(tn.Schema()+"."+tableName,
+			return virtualDefEntry{}, unimplemented.NewWithIssuef(8675,
```
Comment (knz) [inline pkg/sql/virtual_schema.go]: Ditto

File: pkg/sql/vars.go [hunk 2]
```diff
@@ -156,7 +156,7 @@ var varGen = map[string]sessionVar{
 			case "utf8", "unicode", "cp65001":
 				return nil
 			default:
-				return unimplemented.Newf("client_encoding "+encoding,
```
Comment (knz) [inline pkg/sql/vars.go]: Wait a minute can you please extend this to preserve the telemetry.

File: pkg/sql/sqlbase/structured.go [hunk 3]
```diff
@@ -2634,11 +2634,11 @@ func (desc *MutableTableDescriptor) DropConstraint(
 
 	case ConstraintTypeFK:
 		if detail.FK.Validity == ConstraintValidity_Validating {
-			return unimplemented.Newf("drop-constraint-fk-mutation",
+			return unimplemented.NewWithIssuef(42844,
```
Comment (knz) [inline pkg/sql/sqlbase/structured.go]: can you use `WithTelemetry` to differentiate the various FK operations so we get detailed usage telemetry.

**Review by knz — CHANGES_REQUESTED** [review]

File: pkg/sql/virtual_schema.go (hunk 1 above)
Comment (apantel) [reply pkg/sql/virtual_schema.go]: Are the rest okay?

**Review by knz — APPROVED** [review]
top :+1:

**Comment by awoods187** [comment]
Very excited about this work!

**Pull request merged**